crates/factors-derive/src/lib.rs

```diff
@@ -78,7 +77,7 @@ fn expand_factors(input: &DeriveInput) -> syn::Result<TokenStream> {
                         &mut self.#factor_names,
                         #factors_path::InitContext::<Self, #factor_types>::new(
                             linker,
-                            |state| #factors_path::GetFactorState::get::<#factor_types>(state).unwrap(),
+                            |state| &mut state.#factor_names,
```

Comment:
This gets us back to the fast path for most imports, which removes my (tiny, vague) performance concerns.